### PR TITLE
refactor: lightning payment controller

### PR DESCRIPTION
**Pull request by Extheoisah**
This pr adds the following:
1. adds an extra validation check for successful and failed lightning payments using the data returned from the lnd node.
2. introduces the use of sequelize db transactions to ensure the integrity of the data depending of the outcome of the invoice payment
3. adds a new invoice column to the transaction table to store debit payment ln invoices for easy payment tracking.
4. adds a script called create-migration to the Makefile so its easier to create new migrations for the database schema changes.

### CODE REVIEW

File: api/app/helpers/lightning.ts [hunk 1]
```diff
@@ -29,11 +30,49 @@ const payInvoice = async (invoice: string) => {
...
+        );
+        return {
+          success: false,
+          data: null,
+          error: new Error("Payment failed"),
+        };
+      }
     }
   } catch (err) {
     console.error(err);
```
Comment (Emmanuel-Develops) [inline api/app/helpers/lightning.ts:74]: Add a logger for the error including the invoice.

From line38, `JSON.parse(str)` could throw errors for non-standard JSON strings while the payment could have been successful. I was thinking of try/catch but working with malformed objects doesn't seem like a good idea.

```suggestion
    Logger.error(
          `Payment failed for invoice:${invoice} 
          err: ${err?.message ?? err}`
        );
```

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]
Looks good. One nit

File: api/app/helpers/lightning.ts (hunk 1 above)
Comment (Extheoisah) [reply api/app/helpers/lightning.ts:74]: Done!

Comment (Emmanuel-Develops) [reply api/app/helpers/lightning.ts:74]: Nice! You can include the Lightning Invoice in the logger. 
i.e `payment failed for invoice: ${invoice} with err…`

also no need for console.error anymore

**Review by Emmanuel-Develops — APPROVED** [review]

File: api/app/helpers/lightning.ts (hunk 1 above)
Comment (Extheoisah) [reply api/app/helpers/lightning.ts]: Yup! Shows like this in the logs.
```
error:   Payment failed for invoice: lnbc60u1pntyn5tpp5vkupeh4wwe65sa03cgvkyvxnsxj3dty062ut8s0zy06yshpy6qzqdpgw3jhxapqvf6xxgr5wfskuumrwf5hqapqd3shyem9cqzzsxqyz5vqsp5cwj0a2wl2vgu0d4knvklmq6lqq4vukxrqm4k3lw6s2javp0aq72q9qxpqysgqvqzt5d3uc5m33hf4u9k4z4tp5nd7lddldwmaf7acsnwenr3srxwxspdwghwnt2sltr4z3h23aht7786rlp93kmdwf2k92g32sl43l4cqadr0qq
Error: Request failed with status code 409
    at /Users/<user>/bitcoinsearch/transcription-review-backend/api/app/controllers/lightning.controller.ts:95:13
    at Generator.next (<anonymous>)
    at fulfilled (/Users/<user>/bitcoinsearch/transcription-review-backend/api/app/controllers/lightning.controller.ts:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Pull request merged**